sopel/plugins/rules.py

```diff
@@ -732,58 +741,72 @@ def is_unblockable(self) -> bool:
         """
 
     @abc.abstractmethod
-    def is_user_rate_limited(self, nick: Identifier) -> bool:
+    def is_user_rate_limited(
+        self,
+        nick: Identifier,
+        at_time: Optional[datetime.datetime] = None,
+    ) -> bool:
         """Tell when the rule reached the ``nick``'s rate limit.
 
-        :return: ``True`` when the rule reached the limit, ``False`` otherwise
+        :param nick: the nick associated with this check
+        :param at_time: optional aware datetime for the rate limit check
+                        if not given, ``utcnow`` will be used
```

Comment:
Uh-oh, I found a missing punctuation.

```suggestion
        :param at_time: optional aware datetime for the rate limit check;
                        if not given, ``utcnow`` will be used
```